### PR TITLE
feat: implement JsonRpc server

### PR DESCRIPTION
**Pull request by fbozic**
- Implement "simple" JsonRpc server using plain axum
- Integrate server with the rest of the node

### CODE REVIEW

File: crates/primitives/src/server.rs [hunk 1]
```diff
@@ -55,3 +55,81 @@ pub enum WsCommand {
...
+    pub app_id: String,
+    pub method: String,
+    pub args_json: serde_json::Value,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(untagged)]
+pub enum JsonRpcRequestParams {
+    Call(JsonRpcRequestParamsCall),
+}
```
Comment (miraclx) [inline crates/primitives/src/server.rs]:
```suggestion
#[serde(tag = "method", content = "params")]
pub enum JsonRpcRequestPayload {
    Call(JsonRpcRequestParamsCall),
}
```

File: crates/primitives/src/server.rs [hunk 2]
```diff
@@ -55,3 +55,81 @@ pub enum WsCommand {
...
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct JsonRpcRequest {
+    pub jsonrpc: JsonRpcVersion,
+    pub method: String,
+    pub params: Option<JsonRpcRequestParams>,
+    pub id: Option<WsRequestId>,
+}
```
Comment (miraclx) [inline crates/primitives/src/server.rs]:
```suggestion
    pub id: Option<WsRequestId>,
    #[serde(flatten)]
    pub payload: JsonRpcRequestPayload,
}
```

File: crates/primitives/src/server.rs [hunk 3]
```diff
@@ -55,3 +55,81 @@ pub enum WsCommand {
...
+    pub code: u64,
+    pub message: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct JsonRpcResponse {
+    pub jsonrpc: JsonRpcVersion,
+    pub result: Option<JsonRpcResponseResult>,
+    pub error: Option<JsonRpcResponseError>,
```
Comment (miraclx) [inline crates/primitives/src/server.rs]: let's merge these by using a custom enum, take advantage of compiler-backed prohibitions as opposed to developer-informed ones

File: crates/server/src/jsonrpc.rs [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+struct ServiceState {
+    server_sender: ServerSender,
+}
+
+pub(crate) fn service(
+    config: &crate::config::ServerConfig,
+    server_sender: ServerSender,
+) -> eyre::Result<Option<(&'static str, MethodRouter)>> {
+    let path = "/jsonrpc"; // todo! source from config
```
Comment (miraclx) [inline crates/server/src/jsonrpc.rs:37]: you're not checking if `server.jsonrpc.enabled == true`

File: crates/server/src/jsonrpc.rs [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+    Ok(Some((path, post(handle_request).layer(Extension(state)))))
+}
+
+async fn handle_request(
+    Extension(state): Extension<Arc<ServiceState>>,
+    extract::Json(request): extract::Json<calimero_primitives::server::JsonRpcRequest>,
+) -> Json<calimero_primitives::server::JsonRpcResponse> {
+    let result = match request.method.as_str() {
+        "execute" => match request.params {
```
Comment (miraclx) [inline crates/server/src/jsonrpc.rs]: I think a better model in the future would be (considering we'll be optimistically executing transactions)

1. There's only one way to call methods. If at runtime, the method mutates the state in any way, then once the call succeeds, broadcast the transaction. And if not, keep that execution private.
2. But right now, it's okay. Let's maybe use names like `call`, `call_mut`?

But even now, I'd design this differently and model it like meroship instead.

So, you have one enum for all the methods, impl some trait over each of those methods and associate their response & error types.

File: crates/server/Cargo.toml [hunk 6]
```diff
@@ -38,5 +38,6 @@ tokio = { workspace = true, features = ["macros", "rt-multi-thread"] }
 tracing-subscriber = { workspace = true, features = ["env-filter"] }
 
 [features]
+jsonrpc = []
 graphql = ["dep:async-graphql", "dep:async-graphql-axum"]
```
Comment (miraclx) [inline crates/server/Cargo.toml]: it's fine, but:

```suggestion
graphql = ["dep:async-graphql", "dep:async-graphql-axum"]
jsonrpc = []
```

File: crates/primitives/src/server.rs (hunk 3 above)
Comment (fbozic) [reply crates/primitives/src/server.rs]: Done, but probably could be improved more.

File: crates/server/src/jsonrpc.rs (hunk 5 above)
Comment (fbozic) [reply crates/server/src/jsonrpc.rs]: Done (to some extent), but most likely can be improved more.

File: crates/primitives/src/server.rs (hunk 3 above)
Comment (fbozic) [reply crates/primitives/src/server.rs]: Improved more.

File: crates/server/src/jsonrpc.rs (hunk 5 above)
Comment (fbozic) [reply crates/server/src/jsonrpc.rs]: Done with macro all the way (I think) 😄

**Pull request merged**